lib/parse.js

```diff
@@ -107,7 +107,9 @@ var OPERATORS = makePredicate([
     "||"
 ]);
 
-var WHITESPACE_CHARS = makePredicate(characters(" \u00a0\n\r\t\f\u000b\u200b\u180e\u2000\u2001\u2002\u2003\u2004\u2005\u2006\u2007\u2008\u2009\u200a\u202f\u205f\u3000\uFEFF"));
+var WHITESPACE_CHARS = makePredicate(characters(" \u00a0\n\r\t\f\u000b\u200b\u180e\u2000\u2001\u2002\u2003\u2004\u2005\u2006\u2007\u2008\u2009\u200a\u2028\u2029\u202f\u205f\u3000\uFEFF"));
+
+var NEWLINE_CHARS = makePredicate(characters("\r\n\u2028\u2029"));
```

Comment:
Not sure how much science this bench will include (only tested it on my machine on node):

https://gist.github.com/avdg/f176acf2573d15b626c1d2d8ee595144

It seems to me that makePredicate is actually the fastest, followed by the regex method, then followed by String.prototype.indexOf with Array.prototype.includes not much further behind (but might still be more optimizable?)